api/proto/index/v2/index.proto

```diff
@@ -0,0 +1,163 @@
+  repeated Record records = 6;
+  // rankMultiplier must be >= 1.0 and defines the priority of the records if the parser is used or no this field in the records list
+  float rankMultiplier = 7;
+}
+
+// CreateIndexStreamRequest wraps CreateIndexRequest OR a chunk of body stream
+message CreateIndexStreamRequest {
+  // CreateRecordsRequest must be in the first packet of the gRPC stream
+  CreateRecordsRequest meta = 1;
+  // data is the body data, may start from the first packet of the stream
+  bytes data = 2;
+}
+
+// CreateRecordsResult describes the result of the Create call
+message CreateRecordsResult {
+  Nodes nodesCreated = 1;
+  int64 recordsCreated = 2;
+}
+
+// Record represents an index record
+message Record {
+  // id of the record within the node. This value is created by parser or during the records insertion.
+  // it is used for the updating records or deleting them within a node
+  string id = 1;
+  // segment contains the searchable text for the record
+  string segment = 2;
+  // vector is the list of the byte values (stringified) ordered according the basis fields definition
+  bytes vector = 3;
+  // format of the record
+  string format = 4;
+  // rankMultiplier must be >= 1.0 and defines the priority of the records if the parser is used or no this field in the records list
+  float rankMultiplier = 5;
+}
+
+// ListRequest describes input parameters for the list operation
+message ListRequest {
+  // path to the node, for which records to be listed
+  string path = 1;
+  // format contains filter for specific records format
+  optional string format = 2;
+  optional string pageId = 3;
+  optional google.protobuf.Timestamp createdAfter = 4;
+  optional google.protobuf.Timestamp createdBefore = 5;
+  optional int64 limit = 6;
+}
+
+message ListRecordsResult {
+  repeated Record records = 1;
+  optional string nextPageId = 2;
+  int64 total = 3;
+}
+
+// PatchRecordsRequest describes input parameters for index records patch operation.
+message PatchRecordsRequest {
+  // path is the path to the node for the patch request
+  string path = 1;
+  // upsertRecords contains the list of records that should be inserted or updated
+  repeated Record upsertRecords = 2;
+  // deleteRecords contains the list of records that should be deleted
+  repeated Record deleteRecords = 3;
+}
+
+// PatchRecordsResult describes the result of the patch index's records operation
+message PatchRecordsResult {
+  int64 upserted = 1;
+  int64 deleted = 2;
+}
+
+// SearchRecordsRequest describes input parameters for the Search over indexes operation.
+message SearchRecordsRequest {
+  string text = 1;
+  map<string, string> tags = 2;
+  // path contains the path where to run the search. It will be run against all the records in the path sub-tree
+  string path = 3;
+  // strict flag limits the search by the path node only. So the search will not be run in the childrens, but in
+  // th node index records only
+  optional bool strict = 4;
```

Comment:
Is it for searching in the current folder only?

Reply:
In the node only. Only the node records, but nothing else

Reply:
> In the node only. Only the node records, but nothing else

Ok, just I can't imagine how this can be related to a document node. I mean a document node is already a leaf.